nntrainer/models/neuralnet.h

```diff
@@ -479,6 +480,7 @@ class NeuralNetwork : public ml::train::Model {
     swap(lhs.layer_names, rhs.layer_names);
     swap(lhs.def_name_count, rhs.def_name_count);
     swap(lhs.loadedFromConfig, rhs.loadedFromConfig);
+    swap(lhs.manager, rhs.manager);
```

Comment:
is this intended? this does not copy but rather binds the memory of the two models together.

Reply:
Yes, this will allow sharing the memory between model. I think manager should be unique_ptr, naturally model(or more specifically, graph) shouldn't be copy constructible but movable.

This will require bigger change so I decided to go with shared_ptr for now.

Reply:
I think, in order to share the memory manager between models,  there needs to be more careful changes. As far as I understand, it is not considered for now. currently, one model has one manager. and I'm not sure some reason to share the memory manager though.

Reply:
Yes. This will be a unique ptr and the Model won't be copy constructible (but movable), this requires some bigger change so I'll leave it as an issue and do this in later PR.